### PR TITLE
Creating ElastiCache Redis Replication Group composition for Upbound provider

### PR DESCRIPTION
**Pull request by desaidipen**
### What does this PR do?
Creates a ReplicationGroup for ElastiCache Redis cluster.   Along with cluster, composition also creates SubnetGroup, SecurityGroup, SecurityGroup Rule and CloudWatch LogGroups.

### Motivation
Contributing to open source community.

### More

- [X] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)

- [X] Yes, I have added a new example under [examples](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/examples) to support my PR

- [ ] Yes, I have updated the [docs](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/doc) for this feature

- [ ] Yes, I have linked to an issue or feature request (applicable to PRs that solves a bug or a feature request)

**Note**:
 - Not all the PRs require examples and docs
 - We prefer small, well tested pull requests. Please ensure your pull requests are self-contained, and commits are squashed

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: examples/upbound-aws-provider/composite-resources/elasticache-redis.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    subnetIds:
+      - "subnet-001"
+      - "subnet-002"
+  compositionSelector:
+    matchLabels:
+      provider: aws
+      environment: dev
+      configuration: standard
+  resourceConfig:
+    providerConfigName: provider-config-irsa
```
Comment (candonov) [inline examples/upbound-aws-provider/composite-resources/elasticache-redis.yaml:18]:
```suggestion
    providerConfigName: aws-provider-config
```

File: compositions/upbound-aws-provider/elasticache/elasticache-redis.yaml [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+              environment: dev
+              application: my-app
+              snapshotName:
+              from: composition
+            snapshotRetentionLimit: 0
+            snapshotWindow: "09:00-10:00"
+            applyImmediately: true
+            autoMinorVersionUpgrade: "true"
+            maintenanceWindow: "mon:10:40-mon:11:40"
+            notificationTopicArn: arn:aws:sns:us-east-1:791573251752:doNotUse
```
Comment (candonov) [inline compositions/upbound-aws-provider/elasticache/elasticache-redis.yaml:95]: What is this sns topic used for?

File: compositions/upbound-aws-provider/elasticache/definition.yaml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                    type: string
+                  logGroup:
+                    type: object
+                    properties:
+                      retentionInDays:
+                        type: integer
+                required:
+                - region
+                - providerConfigName
+                - notificationTopicArn
```
Comment (candonov) [inline compositions/upbound-aws-provider/elasticache/definition.yaml:95]: notificationTopicArn is not required, remove line

File: compositions/upbound-aws-provider/elasticache/elasticache-redis.yaml (hunk 2 above)
Comment (candonov) [reply compositions/upbound-aws-provider/elasticache/elasticache-redis.yaml:95]: remove the default from the composition

File: examples/upbound-aws-provider/composite-resources/elasticache-redis.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      cluster: crossplane
+      namespace: team-a
+      environment: dev
+      application: my-app-1
+    snapshotRetentionLimit: 7
+    snapshotWindow: "04:00-06:00"
+    applyImmediately: true
+    autoMinorVersionUpgrade: "true"
+    maintenanceWindow: "tue:08:00-tue:10:00"
+    notificationTopicArn: arn:aws:sns:us-east-1:791573251752:sampleNotification
```
Comment (candonov) [inline examples/upbound-aws-provider/composite-resources/elasticache-redis.yaml:31]:
```suggestion
    # optionally provide notificationTopicArn
    notificationTopicArn: arn:aws:sns:us-east-1:111122223333:sampleNotification
```

File: compositions/upbound-aws-provider/elasticache/elasticache-redis.yaml [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+              namespace: team-a
+              environment: dev
+              application: my-app
+              snapshotName:
+              from: composition
+            snapshotRetentionLimit: 0
+            snapshotWindow: "09:00-10:00"
+            applyImmediately: true
+            autoMinorVersionUpgrade: "true"
+            maintenanceWindow: "mon:10:40-mon:11:40"
```
Comment (candonov) [inline compositions/upbound-aws-provider/elasticache/elasticache-redis.yaml:94]: Can you move these defaults to the `definition.yaml` and then in the claim add a comment that there are defaults for there, and you can override them?

File: compositions/upbound-aws-provider/elasticache/definition.yaml [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+                type: object
+                properties:
+                  subnetGroupName:
+                    type: string
+                  vpcId:
+                    type: string
+                  subnetIds:
+                    type: array
+                    items:
+                      type: string
```
Comment (candonov) [inline compositions/upbound-aws-provider/elasticache/definition.yaml:40]:
```suggestion
                    items:
                      type: string
                    required:
                    - subnetIds
```

File: compositions/upbound-aws-provider/elasticache/definition.yaml [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+                    default: default.redis7.cluster.on
+                  logGroup:
+                    type: object
+                    properties:
+                      retentionInDays:
+                        type: integer
+                        default: 14
+                required:
+                - region
+                - subnetIds
```
Comment (candonov) [inline compositions/upbound-aws-provider/elasticache/definition.yaml:102]:
```suggestion
```

**Comment by nabuskey** [comment]
Closing this. If you are still working on this and ready to be reviewed again, please feel free to open it.

**Pull request closed without merging**